src/lwc/geFormRenderer/geFormRenderer.js

```diff
@@ -263,12 +275,143 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
+     * @param fieldWrapper - Array, field ui-label and value using field-api-name as key
+     * @param sectionsList - Array, all sections
+     * @returns {boolean} - true if error message was generated, false if otherwise
+     */
+    getDonorTypeValidationError( fieldWrapper, sectionsList ){
+
+        // get data import record helper
+        const di_record = this.getDataImportHelper( fieldWrapper );
+
+        // donation donor validation depending on selection and field presence
+        let isError = (di_record.donationDonorValue===DONATION_DONOR.isAccount1) ?
+                        di_record.isAccount1Present && di_record.isAccount1ImportedEmpty && di_record.isAccount1NameEmpty :
+                            di_record.donationDonorValue===DONATION_DONOR.isContact1 &&
+                                di_record.isContact1Present && di_record.isContact1ImportedEmpty && di_record.isContact1LastNameEmpty;
+
+        // process error notification when error
+        if (isError) {
+            // highlight validation fields
+            this.highlightValidationErrorFields( di_record, sectionsList, ' ' );
+            // set page error
+            this.hasPageLevelError = true;
+            this.pageLevelErrorMessageList = [ {index: 0, errorMessage: this.getDonationDonorErrorLabel( di_record, fieldWrapper )} ];
+        }
+
+        return isError;
+    }
+
+    /**
+     * Set donation donor error message using custom label depending on field presence
+     * @param diRecord, Object - helper obj
+     * @param fieldWrapper, Array of fields with Values and Labels
+     * @returns {String}, formatted error message for donation donor validation
+     */
+    getDonationDonorErrorLabel( diRecord, fieldWrapper ){
+
+        // init array replacement for custom label
+        let validationErrorLabelReplacements = [ diRecord.donationDonorValue, diRecord.donationDonorLabel ];
+
+        if (diRecord.donationDonorValue === DONATION_DONOR.isAccount1){
+            if (diRecord.isAccount1ImportedPresent)
+                validationErrorLabelReplacements.push(fieldWrapper[DONATION_DONOR_FIELDS.account1ImportedField].label);
+            if (diRecord.isAccount1NamePresent)
+                validationErrorLabelReplacements.push(fieldWrapper[DONATION_DONOR_FIELDS.account1NameField].label);
+        } else {
+            if (diRecord.isContact1ImportedPresent)
+                validationErrorLabelReplacements.push(fieldWrapper[DONATION_DONOR_FIELDS.contact1ImportedField].label);
+            if (diRecord.isContact1LastNamePresent)
+                validationErrorLabelReplacements.push(fieldWrapper[DONATION_DONOR_FIELDS.contact1LastNameField].label);
+        }
+
+        // set message using replacement array - set label depending fields present on template
+        return validationErrorLabelReplacements.length===3 ? format(geErrorDonorTypeValidationSingle, validationErrorLabelReplacements) :
```

Comment:
Let's define a constant instead of using '3'.

Reply:
gotcha